addons/website_sale_loyalty/views/website_sale_templates.xml

```diff
@@ -0,0 +1,150 @@
+                    <td class="text-xl-right border-0"><span t-field="website_sale_order.spent_loyalty_points" /></td>
+                </tr>
+                <t t-if="0 > website_sale_order.available_loyalty_points">
+                    <tr class="text-info">
+                        <td class="text-right border-0">Usable Points:</td>
+                        <td class="text-xl-right border-0"><span t-field="website_sale_order.partner_id.loyalty_points" /></td>
+                    </tr>
+                    <tr class="text-warning">
+                        <td class="text-right border-0">Insufficient Points:</td>
+                        <td class="text-xl-right border-0"><span t-field="website_sale_order.available_loyalty_points" /></td>
+                    </tr>
+                    <tr>
+                        <td colspan="2" class="border-0">
+                            <div class="alert alert-warning">
+                                Some rewards must be removed from cart
+                            </div>
+                        </td>
+                    </tr>
+                </t>
+                <tr t-if="website_sale_order.won_loyalty_points > 0">
+                    <td class="text-right border-0">Points Won:</td>
+                    <td class="text-xl-right border-0"><span t-field="website_sale_order.won_loyalty_points" /></td>
+                </tr>
+                <tr t-if="website_sale_order.won_loyalty_points > 0 or website_sale_order.spent_loyalty_points > 0">
+                    <td class="text-right border-0">Total Points:</td>
+                    <td class="text-xl-right border-0"><span t-field="website_sale_order.reached_loyalty_points" /></td>
+                </tr>
+            </table>
+        </xpath>
+    </template>
+
+    <template id="cart_summary_loyalty" name="Loyalty cart summary" inherit_id="website_sale.cart_summary">
+        <xpath expr="//tbody/tr/td[hasclass('td-img')]" position="replace">
+            <td class='td-img text-center' t-if="line.loyalty_reward_id.image_128">
+                <span><img t-att-src="image_data_uri(line.loyalty_reward_id.image_128)" class="img o_image_64_max rounded" t-att-alt="line.name_short"/></span>
+            </td>
+            <t t-else="">$0</t>
+        </xpath>
+    </template>
+
+    <template id="short_cart_summary_loyalty" name="Loyalty short cart summary" inherit_id="website_sale.short_cart_summary">
+        <xpath expr="//a[@href='/shop/checkout?express=1']" position="attributes">
+            <attribute name="t-if" add="website_sale_order.available_loyalty_points >= 0" separator=" and "/>
+        </xpath>
+    </template>
+
+    <template id="product_configure_loyalty" name="Loyalty Configure Product" inherit_id="website_sale.product">
+        <xpath expr="//div[@id='product_details']/form" position="attributes">
+            <attribute name="t-if" add="not request.website.contains_discount_reward(product)" separator=" and "/>
```

Comment:
It does not prevent "Add to cart" from being available: the button is available for coupon (but its effect is neutralized because the coupon order lines are re-generated).

Reply:
By `prevent to be shown in eshop`, I meant the product, not the button.
For an enduser (not admin or sales permissions), it is not possible to land on this product page, right? (unpublished + sale_ok False)
If it is not a real use case, it shouldn't be handled IMHO, as the same can be done through RPC.

Reply:
Ok, but then I should remove the link from the cart to the product, which leads to a 403.

Reply:
Now that you mention it, indeed..
I don't think it makes any sense to be able to land on a promotion / coupon / loyalty product, isn't it?